woocommerce/woocommerce-gutenberg-products-block
# Product SKU: Adds support for color, typography, and spacing

This Pull Request enhances the styling capabilities of the Product SKU block by introducing additional support for color, typography & spacing. This improvement provides users with more flexibility when customizing the appearance of the Product SKU block.

Here is what Product SKU block supports now:

**Color:**
- Text color
- Background color

**Typography:**
- Font family 
- Font size
- Font Appearance
- Line height
- Letter spacing 
- Text decoration 
- Letter Case

**Spacing:**
- Margin
- Padding



Part of #8691 & #7954




### User Facing Testing

#### Test using blocks sidebar
- Create a new page or post
- Add the "Products" block to your post.
- Add Product SKU block as an inner block if not included already
- Select the Product SKU block and open the block settings in the right sidebar.
- Customize the block using the new styling options available:
   - Change the text color, background color, font family, font size, font appearance, line height, letter spacing, text decoration, letter case, margin, and padding under the appropriate settings sections.
- Verify that the styling changes are immediately visible and applied correctly within the editor itself.
- Save the post, then view it on the front-end to ensure the styling changes are visible and applied correctly.

#### Test using Global Styles
- Open the Global Styles sidebar by clicking the "Styles" button at the top right corner of the editor.

https://user-images.githubusercontent.com/16707866/228781997-24a9a29f-4be7-46d4-b1b3-41520692d86d.mov

- In the Global Styles sidebar, locate the "Product SKU" block.
- Customize the block using the new styling options available under the "Product SKU" block in the Global Styles sidebar:
   - Change the text color, background color, font family, font size, font appearance, line height, letter spacing, text decoration, letter case, margin, and padding under the appropriate settings sections.
- Verify that these applied correctly within editor & on frontend:
   - Create a new post
   - Add Products block
   - Add Product SKU block as an inner block if not included already
   - Check the Product SKU block in both the editor and on the front-end to ensure that the changes made through the Global styles UI are accurately applied:
       - In the editor, inspect Product SKU block & ensure it matches the values set in the Global styles UI.
       - Save the post & publish it. On frontend, inspect Product SKU block & ensure it matches the values set in the Global styles UI.

> Also, verify the changes in All Products block, which also uses Product SKU block.



### WooCommerce Visibility




### Changelog

> Enhance Product SKU block styling capabilities with additional support for color, typography & spacing
